core/chains/evm/bulletprooftxmanager/eth_broadcaster.go

```diff
@@ -345,20 +359,36 @@ func (eb *EthBroadcaster) handleInProgressEthTx(etx EthTx, attempt EthTxAttempt,
 	}
 	parentCtx := context.TODO()
 
-	if etx.Simulate {
-		simulationCtx, cancel := context.WithTimeout(parentCtx, SimulationTimeout)
-		defer cancel()
-		if b, err := simulateTransaction(simulationCtx, eb.ethClient, attempt, etx); err != nil {
-			if jErr := evmclient.ExtractRPCError(err); jErr != nil {
-				eb.logger.CriticalW("Transaction reverted during simulation", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "rpcErr", jErr.String(), "returnValue", b.String())
-				etx.Error = null.StringFrom(fmt.Sprintf("transaction reverted during simulation: %s", jErr.String()))
-				return eb.saveFatallyErroredTransaction(&etx)
-			}
-			eb.logger.Warnw("Transaction simulation failed, will attempt to send anyway", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "err", err, "returnValue", b.String())
-		} else {
-			eb.logger.Debugw("Transaction simulation succeeded", "ethTxAttemptID", attempt.ID, "txHash", attempt.Hash, "returnValue", b.String())
-		}
+	checkerSpec, err := etx.GetChecker()
+	if err != nil {
+		return errors.Wrap(err, "parsing transmit checker")
+	}
+
+	checker, err := eb.checkerFactory.BuildChecker(checkerSpec)
+	if err != nil {
+		return errors.Wrap(err, "building transmit checker")
+	}
+
+	// If the transmit check does not complete within 2 seconds, the transaction will be sent
+	// anyway.
+	checkCtx, cancel := context.WithTimeout(parentCtx, 2*time.Second)
+	defer cancel()
+	err = checker.Check(checkCtx, eb.logger, etx, attempt)
+	if errors.Is(err, context.Canceled) {
+		eb.logger.Errorw("Transmission checker timed out, sending anyway",
+			"ethTxId", etx.ID,
+			"meta", etx.Meta,
+			"checker", etx.TransmitChecker)
+	} else if err != nil {
+		etx.Error = null.StringFrom(err.Error())
+		eb.logger.Infow("Transmission checker failed, fatally erroring transaction.",
+			"ethTxId", etx.ID,
+			"meta", etx.Meta,
+			"checker", etx.TransmitChecker,
+			"err", err)
+		return eb.saveFatallyErroredTransaction(&etx)
 	}
+	cancel()
```

Comment:
I don't think this explicit `cancel()` call is necessary here, but could be wrong. By virtue of arriving at this line, the context is either already cancelled or the Check() call completed with an error/nil error, so it's a bit moot. The `defer` is enough I think.

Reply:
I don't think it matters in practice, but from context.go:

`code should call cancel as soon as the operations running in this Context complete`

I went ahead and called it here so that the context doesn't stay open during the rest of handleInProgressEthTx  (but again I don't think it makes much of a difference, agree it looks a bit awkward..)

Reply:
Doesn't this mean we might double cancel since we already deferred it?

Reply:
Yeah we will double cancel assuming we don't abort, but double cancel is fine according to context.go (subsequent cancels do nothing)